poetry/repositories/legacy_repository.py

```diff
@@ -359,10 +359,13 @@ def _get_release_info(self, name, version):  # type: (str, str) -> dict
 
         return data.asdict()
 
-    def _get(self, endpoint):  # type: (str) -> Union[Page, None]
+    def _get(self, endpoint):  # type: (str) -> Optional[Page]
         url = self._url + endpoint
         response = self.session.get(url)
         if response.status_code == 404:
             return
-
+        if response.status_code != 200:
```

Comment:
This could cause some unintended side-effects. I would suggest we log a debug message for now as a start.

Reply:
1) Adding debug seems a to low level, maybe warning on 401 and debug otherwise?
2) Does adding debug logging require a test?
3) How could we get a clear picture on the possible unintended side-effects, maybe we could find and fix those?

Reply:
1. If the code is `>=400 and <500`, warning/info should be okay I guess, for everything else a debug should be fine. However, the intention here would be to assist in debugging why something does not work, this would mean executing the command with `-v` if it fails.
2. If it is trivial, I would say just post the cli output of your manual verification in the PR description could do the trick. A test would be great though, because you can use pytest to intercept and veriffy logging along with mocking a response. This allows us to catch any regression later on.
3. That might be difficult difficult; since legacy repository implementations are very wild west unfortunately. The side-effect might be at run-time for end-users using a non-standard repository. For example, we have seen different response codes for success from artifactory/azure private indices etc. Hence why I am reluctant to raise an error on `!=200`. We do not want to break any working scenarios.

Reply:
1) Will update my PR to reflect 1 (warning for 400 to 500 range) and debug otherwise
2) Ok, will add a mock test to cover the log line
3) Ah, shame. I can understand it's a wild west, maybe we should introduce configuration to pick the backend implementation of an source at some point or maybe we could directly hook into some pip code for repository handling at a later date.

I will work on 1 and 2 and resolve this discussion at the time those changes are in the PR. Thank you for responding.